### DISCUSSION TITLE
Pytorch reserved memory issue

### DISCUSSION THREAD

**Discussion by shlabada** [nateraw/stable-diffusion-videos#133] [Q&A] [answered]
I have the same error message everytime I try to execute the stable-diffusion-videos code on my computer.
**RuntimeError: CUDA out of memory. Tried to allocate 2.00 MiB (GPU 0; 6.00 GiB total capacity; 4.31 GiB already allocated; 844.80 KiB free; 4.71 GiB reserved in total by PyTorch)**
I didn't found answers on the internet so i'm asking here. thank you in advance.

**Comment by nateraw** [comment, marked as answer]
You've run out of GPU memory. Looks like you have 6GB available...not sure this will be enough. But, you might be able to make it work with attention slicing...call `.enable_attention_slicing()` on the pipeline right after you initialize it  and then try again to see if you're able to make it work

Also make sure to try with `batch_size=1` for minimal example.

```python
from stable_diffusion_videos import StableDiffusionWalkPipeline

from diffusers.models import AutoencoderKL
from diffusers.schedulers import LMSDiscreteScheduler
import torch

pipe = StableDiffusionWalkPipeline.from_pretrained(
    'runwayml/stable-diffusion-v1-5',
    vae=AutoencoderKL.from_pretrained(f"stabilityai/sd-vae-ft-ema"),
    torch_dtype=torch.float16,
    revision="fp16",
    safety_checker=None,
    scheduler=LMSDiscreteScheduler(
        beta_start=0.00085, beta_end=0.012, beta_schedule="scaled_linear"
    )
).to("cuda")

pipe.enable_attention_slicing()

# ...generate images...
```

**Comment by shlabada** [comment]
The safety  checker is desactivated, can I bypass this error ? or how can I turn the safety checker on ?

**You have disabled the safety checker for <class 'stable_diffusion_videos.stable_diffusion_pipeline.StableDiffusionWalkPipeline'> by passing `safety_checker=None`. Ensure that you abide to the conditions of the Stable Diffusion license and do not expose unfiltered results in services or applications open to the public. Both the diffusers team and Hugging Face strongly recommend to keep the safety filter enabled in all public facing circumstances, disabling it only for use-cases that involve analyzing network behavior or auditing its results. For more information, please have a look at https://github.com/huggingface/diffusers/pull/254 .**

**Reply by nateraw** [reply]
You just would not pass the safety checker arg...should initialize as expected then. I find the safety checker has a lot of false positives when generating videos, which is why I turn it off

**Reply by shlabada** [reply]
Thank you very much, working great !

**Comment by shlabada** [comment]
Do you know if your solution works with other stable diffusion based video generator like Deforum ?
